### PR TITLE
ci:  update the commitlint rules and version

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR does below items

* Uses trailer-exists instead of signed-off-by to verify the sign-off-by message
* Updates the mergify rules to consider the commitlint status.
* Update commitlint version to latest.

Got suggestion from https://github.com/conventional-changelog/commitlint/issues/2704#issuecomment-892144091

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: build.env [hunk 1]
```diff
@@ -20,7 +20,7 @@ GOLANG_VERSION=1.16.4
 GO111MODULE=on
 
 # commitlint version
-COMMITLINT_VERSION=12.1.4
+COMMITLINT_VERSION=13.1.0
```
Comment (nixpanic) [inline build.env]: Is it possible to use a tag like `latest`? We would like to use the most current version by default,  until it breaks :-)

Comment (Madhu-1) [reply build.env]: Yes updated

**Comment by nixpanic** [comment]
Would you mind updating the commit `ci: trailer-exists to verify sign-off` so that you have two lines like:

```
Signed-off-by: Madhu
Suggested-by: Ade Attwood
```

With that, the commitlint change will tested :smile:

**Comment by Madhu-1** [comment]
> ```
> Suggested-by: Ade Attwood
> ```

Done

**Review by nixpanic — DISMISSED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by Madhu-1** [comment]
This needs a manual merge.

**Review by pkalever — APPROVED** [review]
LGTM

**Comment by Madhu-1** [comment]
Merging it manually as its a accepted solution.

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Pull request merged**